core/src/main/java/com/segment/analytics/kotlin/core/State.kt

```diff
@@ -25,14 +25,38 @@ data class System(
 
     companion object {
         fun defaultState(configuration: Configuration, storage: Storage): System {
-            val settings = try {
-                Json.decodeFromString(
-                    Settings.serializer(),
-                    storage.read(Storage.Constants.Settings) ?: ""
-                )
-            } catch (ignored: Exception) {
-                configuration.defaultSettings
+            val storedSettings = storage.read(Storage.Constants.Settings)
+            val defaultSettings = configuration.defaultSettings ?: Settings(
+                integrations = buildJsonObject {
+                    put(
+                        "Segment.io",
```

Comment:
maybe better to make this a constant in SegmentDestination and reference it here?

Reply:
I think we can add constructors in Settings so it's consistent with [swift's implementation](https://github.com/segmentio/analytics-swift/blob/main/Sources/Segment/Settings.swift#L18) and that forces every settings created to have a segment destination